config/detect.yaml

```diff
@@ -0,0 +1,456 @@
+  #      - /var/log/*http*/*.log
+
+  apache2-systemd-deb:
+    when:
+      - UnitFound("apache2.service")
+      - PathExists("/etc/debian_version")
+    install:
+      collections:
+        - crowdsecurity/apache2
+    datasource:
+      filenames:
+        - /var/log/apache2/*.log
+      labels:
+        type: apache2
+
+  apache2-systemd-rpm:
+    when:
+      - UnitFound("httpd.service")
+      - PathExists("/etc/redhat-release")
+    install:
+      collections:
+        - crowdsecurity/apache2
+    datasource:
+      filenames:
+        - /var/log/httpd/*.log
+        # XXX /var/log/*http*/*.log
+      labels:
+        type: apache2
+
+  #
+  # crowdsecurity/asterisk
+  #
+
+  asterisk-systemd:
+    when:
+      - UnitFound("asterisk.service")
+    install:
+      collections:
+        - crowdsecurity/asterisk
+    datasource:
+      labels:
+        type: asterisk
+      filenames:
+        - /var/log/asterisk/*.log
+
+  #
+  # crowdsecurity/caddy
+  #
+
+  caddy-systemd:
+    when:
+      - UnitFound("caddy.service")
+    install:
+      collections:
+        - crowdsecurity/caddy
+    datasource:
+      labels:
+        type: caddy
+      filenames:
+        - /var/log/caddy/*.log
+
+  #
+  # crowdsecurity/dovecot
+  #
+
+  dovecot-systemd:
+    when:
+      - UnitFound("dovecot.service")
+    install:
+      collections:
+        - crowdsecurity/dovecot
+    datasource:
+      labels:
+        type: syslog
+      filenames:
+        - /var/log/mail.log
+
+  #
+  # LePresidente/emby
+  #
+
+  emby-systemd:
+    when:
+      - UnitFound("emby-server.service")
+    install:
+      collections:
+        - LePresidente/emby
+    datasource:
+      labels:
+        type: emby
+      filenames:
+        - /var/log/embyserver.txt
+
+  #
+  # crowdsecurity/endlessh
+  #
+
+  endlessh-systemd:
+    when:
+      - UnitFound("endlessh.service")
+    install:
+      collections:
+        - crowdsecurity/endlessh
+    datasource:
+      source: journalctl
+      labels:
+        type: syslog
+      # XXX this? or /var/log/syslog?
+      journalctl_filter:
+        - "_SYSTEMD_UNIT=endlessh.service"
+
+  #
+  # crowdsecurity/gitea
+  #
+
+  # XXX untested
+
+  gitea-systemd:
+    when:
+      - UnitFound("gitea.service")
+    install:
+      collections:
+        - crowdsecurity/gitea
+    datasource:
+      labels:
+        type: gitea
+      filenames:
+        - /var/log/gitea.log
+
+  #
+  # crowdsecurity/haproxy
+  #
+
+  haproxy-systemd:
+    when:
+      - UnitFound("haproxy.service")
+    install:
+      collections:
+        - crowdsecurity/haproxy
+    datasource:
+      labels:
+        type: haproxy
+      filenames:
+        - /var/log/haproxy/*.log
+
+  #
+  # firewallservices/lemonldap-ng
+  #
+
+  lemonldap-ng-systemd:
+    when:
+      - UnitFound("lemonldap-ng-fastcgi-server.service")
+    install:
+      collections:
+        - firewallservices/lemonldap-ng
+    datasource:
+      # XXX todo where are the logs?
```

Comment:
We need to find this, as this won't pass the validator if file path is missing.

From https://lemonldap-ng.org/documentation/2.0/logs.html#default-loggers  it seems it uses same location as Apache ? 

Do you think maybe we could make the validator pass in such situations while emitting a warning ?

Reply:
The datasource of each application must be verified manually before going in production, since I did some guesswork. In the bats-detect directory you find the tests to install on .deb, .rpm platforms (I deploy them in ansible fist) and check the detection worked correctly, but logs are not automatically verified.
If we are not sure about a datasource, we can just skip the whole section. I'm not sure warnings are useful with apt/dnf, but we could collect them with something like:

```
 - when: blabla
   install:
      collections: bla/bla
   message: |
      The collection bla/bla was installed, but it has no datasource. Enjoy!
```

Then it would be printed together at the end.

We could even use the mechanism to have parts with when/message and nothing else. Could be useful.
Or maybe allow an empty datasource if there's a comment: field, which would go in its acquis.d/*.yaml file